ADRs/RH/SIG-SRE/ADR-00002 Open Source Documentation.md

```diff
@@ -0,0 +1,59 @@
+
+## Goals
+
+* Propose a model for maintaining 2 versions of documents, for upstream and downstream consumption
+* Provide a worked example of this model as a reference
+
+## Non-goals
+
+* Implement a complete solution for a product or service
+
+## Current Architectures
+
+A Standard Operating Procedure (SOP) is a document used by SREs in response to alerts and other events.
+An example of what a current SOP architecture looks like is the Red Hat OpenShift Streams for Apache Kafka (RHOSAK) managed service. The RHOSAK SOPs are located in a private repository, as the SOPs contain references to Red Hat specific internal information that is not suitable for consumption by external users. Such information includes links to Vault secrets, information about how RHOSAK specifically set up their stage and production environments and information on the infrastructure itself, for example what certain custom resources should look like. The flow of how information is contributed to the RHOSAK SOPs is for an engineer or SRE to submit PRs to the repo.
+
+## Proposed Architecture
+
+The proposed solution is to follow a two repository model. One private repository that will contain SOPs that have the Red Hat internal information mentioned above, the other being a public repository that will have SOPs with the Red Hat internal information removed. This model will allow the SRE teams to use the private SOP repository without any information loss and allow upstream users to use and contribute to the SOP without breaching any Red Hat data policies.
+
+![Upstream, Downstream Pattern](images/upstream_downstream_docs_pattern.png)
+
+The content of the SOPs will be mirrored/synced between both repositories apart from the Red Hat internal information. The flow of contributing information to the repositories can be done both ways i.e. from downstream to upstream and from upstream to downstream. This information can be contributed manually through the creation of pull requests or it can be done autonomously using the mirroring features in the repository. The team in charge of the SOPs will be responsible for maintaining the information between both repositories. This includes merging contributions from external users in the upstream repository and ensuring, if applicable, it gets added to the downstream repository and vice versa.
+
+SOPs from a private downstream repository that contain sensitive information, will have to be cleaned of all sensitive information before being pushed to the upstream public repository. Such information might be links to vault secrets or information about the managed service infrastructure etc. The pre-existing Structure of  the SOPs should also be looked at, to see if any information can be removed that is no longer necessary or can be moved to a different internal location, this will help reduce overhead when syncing.
```

Comment:
From the [Open Source requirements for Services]:
> - A public contributor can use the same workflow as a typical team member to make a change to the service.

If a "change to the service" would also include changes to its documentation or runbooks, this would mean that the "main" documentation repository should be the upstream one. The downstream one should then only be used for changes of internal details.

[Open Source requirements for Services]: https://www.operate-first.cloud/community/open-source-services.html

```suggestion
The proposed solution is to follow a two repository model. The public repository contains the SOPs without any company internal information, and is the main documentation repository. The internal repository contains (sensitive) company internal information.

For changes that only contain public information, only one change request to the public repository is needed. For changes that also contain internal information, two change requests are needed: one to the public repository with the public information, and one to the internal repository with the internal information. To enable successful merging of the information from both repositories, public and internal information should not be intermingled in any one file.

![Upstream, Downstream Pattern](images/upstream_downstream_docs_pattern.png)

SRE teams need to be able to use SOPs with the internal information included. With the proposed repository layout, this could be implemented in multiple ways, e.g.:
- all changes to the public repository are synchronized to the internal repository, so the internal repository by itself will contain the SOPs with internal information included
- during a publishing step, two versions of of the final documentation are generated: (1) a public (upstream) version, only containing information from the public repository, and an internal (downstream) version, containing information from both the public and internal repositories
```

Reply:
> If a "change to the service" would also include changes to its documentation or runbooks, this would mean that the "main" documentation repository should be the upstream one. The downstream one should then only be used for changes of internal details.

@mh21 agreed.
After careful review of the feedback here & in the gdoc internally that this ADR is based on, I'm going to close this PR and open a new one that takes a more risk averse approach that should match what you're saying.
(The internal link for the new ADR is https://docs.google.com/document/d/11o88BcDLAGOHCl3Lo3F7OM6bvQLhDvH7gUreVIImnq8/edit , which I intend to make a generic version of for here)